Fix Acceptance Test For `TestProvider_insecure`

Acceptance test runs have been excluding [`TestProvider_insecure`](https://github.com/terraform-providers/terraform-provider-github/blob/master/github/provider_test.go#L147) in error.  At some point, this test began failing and is currently failing for the `master` branch.

```console
 TF_ACC=1 go test -v -timeout 30m  ./... -run TestProvider_insecure
?   	github.com/terraform-providers/terraform-provider-github	[no test files]
=== RUN   TestProvider_insecure
--- FAIL: TestProvider_insecure (0.21s)
    testing.go:654: Step 1 error: errors during refresh:

        Error: GET https://localhost:59070/orgs/terraformtesting: 404  []

          on /var/folders/t1/p9cyhv6s2wg4g8s_x6zkpz_m0000gn/T/tf-test017474600/main.tf line 1:
          (source code not available)


FAIL
FAIL	github.com/terraform-providers/terraform-provider-github/github	0.475s
```

The test validates functionality related to non-GitHub.com deployments as well as a debug mode not recommended for production use cases.

/cc https://github.com/terraform-providers/terraform-provider-github/pull/331

--- a/.github/workflows/acceptance-tests.yml
+++ b/.github/workflows/acceptance-tests.yml
@@ -34,7 +34,7 @@ jobs:
       - name: Generate Test Fixtures
         run: |
           openssl req -x509 -newkey rsa:4096 -days 1 -nodes \
-            -subj "/C=US/ST=CA/L=San Francisco/O=HashiCorp, Inc./CN=localhost" \
+            -subj "/C=US/ST=CA/L=San Francisco/O=HashiCorp, Inc./CN=untrusted" \
             -keyout github/test-fixtures/key.pem -out github/test-fixtures/cert.pem
 
       - name: Acceptance Tests

--- a/github/provider_test.go
+++ b/github/provider_test.go
@@ -70,7 +70,7 @@ func testAccPreCheck(t *testing.T) {
 	}
 }
 
-func TestProvider_individual(t *testing.T) {
+func TestAccProvider_individual(t *testing.T) {
 
 	username := "hashibot"
 	resource.ParallelTest(t, resource.TestCase{
@@ -110,7 +110,7 @@ func TestProvider_individual(t *testing.T) {
 	})
 }
 
-func TestProvider_anonymous(t *testing.T) {
+func TestAccProvider_anonymous(t *testing.T) {
 
 	username := "hashibot"
 	resource.Test(t, resource.TestCase{
@@ -144,7 +144,7 @@ func TestProvider_anonymous(t *testing.T) {
 	})
 }
 
-func TestProvider_insecure(t *testing.T) {
+func TestAccProvider_insecure(t *testing.T) {
 	// Use ephemeral port range (49152–65535)
 	port := fmt.Sprintf("%d", 49152+rand.Intn(16382))
 

--- a/github/provider_test.go
+++ b/github/provider_test.go
@@ -199,6 +199,7 @@ func githubTLSApiMock(port, certFile, keyFile string, t *testing.T) (string, fun
 	mux.HandleFunc("/users/hashibot", testRespondJson(userResponseBody))
 	mux.HandleFunc("/users/hashibot/gpg_keys", testRespondJson(gpgKeysResponseBody))
 	mux.HandleFunc("/users/hashibot/keys", testRespondJson(keysResponseBody))
+	mux.HandleFunc("/orgs/"+testOrganization, testRespondJson(orgResponseBody(port)))
 
 	server := &http.Server{
 		Addr:    ":" + port,
@@ -317,3 +318,14 @@ const keysResponseBody = `[
     "key": "ssh-rsa AAA..."
   }
 ]`
+
+func orgResponseBody(port string) string {
+	url := fmt.Sprintf(`https://localhost:%s/orgs/%s`, port, testOrganization)
+	return fmt.Sprintf(`
+{
+	"login": "%s",
+	"url" : "%s",
+	"repos_url": "%s/repos"
+}
+`, testOrganization, url, url)
+}

